Extension/src/api.ts

```diff
@@ -0,0 +1,31 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as vscode from 'vscode';
+
+export interface CppToolsApi {
+    registerCustomConfigurationProvider(provider: CustomConfigurationProvider): void;

```

Comment:
>registerCustomConfigurationProvider [](start = 4, length = 35)

Not blocking for this PR, but can you add comments in here (for each file/property/interface), especially for the API functions? Because this is the public contract for this extension, it should be well-documented